src/python_testing/TC_CNET_1_4.py

```diff
@@ -0,0 +1,106 @@
+# test-runner-run/run1/script-args: --storage-path admin_storage.json --commissioning-method on-network --discriminator 1234 --passcode 20202021 --PICS src/app/tests/suites/certification/ci-pics-values --trace-to json:${TRACE_TEST_JSON}.json --trace-to perfetto:${TRACE_TEST_PERFETTO}.perfetto
+# === END CI TEST ARGUMENTS ===
+
+import logging
+
+import chip.clusters as Clusters
+from matter_testing_support import MatterBaseTest, TestStep, async_test_body, default_matter_test_main
+from mobly import asserts
+
+
+class TC_CNET_1_4(MatterBaseTest):
+    def steps_TC_CNET_1_4(self):
+        return [TestStep(1, "TH is commissioned", is_commissioning=True),
+                TestStep(2, 'TH performs a wildcard read of Network Commissioning clusters across all endpoints, and save the number of Network Commissioning clusters as `NumNetworkCommissioning` for future use. If `NumNetworkCommissioning` is 0, skip the remaining steps in this test case'),
+                TestStep(3, 'TH reads from the DUT the Descriptor Cluster DeviceTypeList attribute from Endpoint 0, verify that the Network Commissioning cluster id (0x0031) is listed in the ServerList'),
+                TestStep(4, 'TH reads from the DUT the Descriptor Cluster DeviceTypeList attribute on each endpoint (except for Endpoint 0) that hosts a Network Commissioning cluster, verify that the Secondary Network Interface device type id (0x0019) is listed in the DeviceTypeList'),
+                TestStep(5, 'TH reads from the DUT the General Commissioning Cluster SupportsConcurrentConnection attribute if NumNetworkCommissioning is greater than 1, verify that it is true')]
+
+    def def_TC_CNET_1_4(self):
+        return '[TC-CNET-1.4] Verification for Secondary Network Interface [DUT-Server]'
+
+    def pics_TC_CNET_1_4(self):
+        return ['CNET.S']
+
+    # Override default timeout.
+    @property
+    def default_timeout(self) -> int:
+        return 200
+
+    @async_test_body
+    async def test_TC_CNET_1_4(self):
+        # Commissioning is already done
+        self.step(1)
+
+        cnet = Clusters.NetworkCommissioning
+        afeam = cnet.Attributes.FeatureMap
+
+        self.step(2)
+        # Read FeatureMap attribute with wildcard endpoint
+        feature_map_results = await self.default_controller.ReadAttribute(self.dut_node_id, [(afeam)], fabricFiltered=True)
+
+        NumNetworkCommissioning = len(feature_map_results)
+        if NumNetworkCommissioning == 0:
+            logging.info('No endpoint has Network Commissioning Cluster, skipping remaining steps')
+            self.skip_all_remaining_steps(3)
+            return
+
+        self.step(3)
+        cdesc = Clusters.Descriptor
+        aser = cdesc.Attributes.ServerList
+
+        server_list = await self.read_single_attribute_check_success(cluster=Clusters.Descriptor, attribute=aser, endpoint=0)
+        if 49 not in server_list:
```

Comment:
Not that this is going to change, but I assume we have a better way to get the ID than hard code it 🤔